### DISCUSSION TITLE
Where to put losses when using user defined models? What are the correct boundaries/buckets?

### DISCUSSION THREAD

**Discussion by natecostello** [pvlib/pvlib-python#1956] [Q&A] [answered]
I'm using  some user defined functions that populate modelchainresults: mc.results.ac, mc.results.losses, mc.results.dc_ohmic_losses.

I would like to ensure my "book keeping" is consistent with the intent of those result attributes.

1.  Where do DC->AC conversion losses go, or do they not go anywhere?
2.  Are dc_ohmic_losses included in losses? 

For my system I've got Modules->PV-to-MPPT-Wire->MPPT->MPPT-to-Battery-Wire->Battery.  I've got I2R for the wire on each side, and an efficiency model for the MPPT.   I would like to know the appropriate or consensus way to attribute the loss associated with each.

My current leaning is to treat the MPPT loss and I2R from MPPT-to-Battery as "inverter loss" and treat the connection to the battery as the output of the system (what you would get from result.ac.  But I'm not sure where that combined loss would go (if anywhere).  

That would leave the PV side wiring loss, which I think should be captured as result.dc_ohmic, but I'm not sure if that should also go in result.losses.

A simple equation of this = that - this_other_thing would be helpful.

I

**Comment by cwhanse** [comment, marked as answer]
In pvlib's current construction, DC->AC conversion losses are accounted for in the inverter model. There isn't a separate 'loss' function. However, you could include a factor in a user-defined `ModelChain.losses_model`. A user-defined losses model would need to behave like the [pvwatts loss model](https://github.com/pvlib/pvlib-python/blob/6bb149fa8c933ea32a060198864a12887dc5c538/pvlib/modelchain.py#L1244); it would apply the product of factors to the DC power and record the result on `ModelChain.result.dc`.

DC ohmic losses are handled differently - the losses is in units of power and is subtracted from DC power, rather than being a factor multiplied by DC power.

I'll try to formulate, and welcome corrections:

DC power at the inverter = (DC power from array) x (product of losses in losses_model) - (dc ohmic loss)

AC power from inverter = f_inverter (DC power at the inverter)
